File: adventjs/2025/19/solution.js
/**
 * @param {string[][]} routes - Array of [origin, destination] pairs
 * @returns {string[]} The reconstructed route
 */
function revealSantaRoute(routes) {
  if (!routes.length) return [];
  
  const routeMap = new Map(routes.map(r => [r[0], r[1]]));
  const route = [...routes[0]];
  let currentDest = routes[0][1];
  
  while (routeMap.has(currentDest)) {
    currentDest = routeMap.get(currentDest);
    route.push(currentDest);
  }
  
  return route;
}

File: adventjs/2025/19/solution.py
def reveal_santa_route(routes: list[list[str]]) -> list[str]:
  if not routes:
    return []
  
  route_map = {r[0]: r[1] for r in routes}
  route = routes[0][:]
  current_dest = routes[0][1]
  
  while current_dest in route_map:
    current_dest = route_map[current_dest]
    route.append(current_dest)
  
  return route
